src/test/java/org/jenkinsci/plugins/workflow/support/storage/BulkFlowNodeStorageTest.java

```diff
@@ -58,4 +68,37 @@ public final class BulkFlowNodeStorageTest {
         assertThat(ids, is(IntStream.rangeClosed(2, 43).mapToObj(Integer::toString).collect(Collectors.toList())));
     }
 
+    @LocalData
+    @Test public void actionDeserializationShouldBeRobust() throws Exception {
+        /*
+        var p = r.createProject(WorkflowJob.class);
+        p.addProperty(new DurabilityHintJobProperty(FlowDurabilityHint.PERFORMANCE_OPTIMIZED));
+        p.setDefinition(new CpsFlowDefinition(
+                "stage('test') {\n" +
+                "  semaphore('wait')\n" +
+                "}\n", true));
+        var b = p.scheduleBuild2(0).waitForStart();
+        SemaphoreStep.waitForStart("wait/1", b);
+        ((CpsFlowExecution) b.getExecution()).getStorage().flush();
+        Thread.sleep(30*1000);
+        */
+        var p = r.jenkins.getItemByFullName("test0", WorkflowJob.class);
+        var b = p.getLastBuild();
+        var stageBodyStartNode = (StepStartNode) b.getExecution().getNode("4");
+        assertThat(stageBodyStartNode, not(nullValue()));
+        var label = stageBodyStartNode.getPersistentAction(LabelAction.class);
+        assertThat(label.getDisplayName(), equalTo("test"));
```

Comment:
Without the patch, the whole Pipeline itself fails to load because the active head refers to a node with the action from the class that is no longer present.